apps/web/src/pages/api/payment/stripe/[publickey].ts

```diff
@@ -0,0 +1,77 @@
+import { NextApiRequest, NextApiResponse } from "next"
+import { getStripeObject, getStripeWebhookSecret } from "@dotkomonline/api/src/lib/stripe"
+
+import Stripe from "stripe"
+import { createContextInner } from "@dotkomonline/api"
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "POST") {
+    res.status(405).end()
+    return
+  }
+
+  const publicKey = req.query.publickey as string
+  const sig = req.headers["stripe-signature"] as string
+
+  const stripe = getStripeObject(publicKey)
+  const endpointSecret = getStripeWebhookSecret(publicKey)
+
+  if (!stripe || !endpointSecret) {
+    console.warn("No stripe account found for the given public key")
+    res.status(400).end()
+    return
+  }
+
+  let event: Stripe.Event
+  try {
+    const body = await buffer(req)
+    event = stripe.webhooks.constructEvent(body, sig, endpointSecret)
+  } catch (e) {
+    const err = e as Error
+    res.status(400).send(`Webhook Error: ${err.message}`)
+    return
+  }
+
+  if (event.type.startsWith("checkout.session")) {
+    const ctx = await createContextInner({ auth: null })
```

Comment:
huh how do you have access to this in the web folder? If this handler is moved to the API folder, you don't need to import this